Remote resolution inside GIT repo will crash tkn pac resolve

When trying to `tkn pac resolve` with a remote annotation referencing a task
inside the repo, it will crash tkn pac resolve, the yaml snippet : 

```yaml
pipelinesascode.tekton.dev/task: "[git-clone,.tekton/buildah.yaml]"
```

and the crash

```shell
% tkn pac resolve -f ./run.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x140 pc=0x149c8fa]

goroutine 1 [running]:
github.com/openshift-pipelines/pipelines-as-code/pkg/provider/github.(*Provider).GetFileInsideRepo(0xc0001100a8, {0x1a9d690, 0xc000040128}, 0xc000188800, {0xc0001164cb, 0x14}, {0x0, 0x8})
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/provider/github/github.go:106 +0xba
github.com/openshift-pipelines/pipelines-as-code/pkg/matcher.RemoteTasks.getTask({0x183d3a0}, {0x1a9d690, 0xc000040128}, {0x1ac4f00, 0xc0001100a8}, {0xc0001164cb, 0x14})
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/matcher/annotation_tasks_install.go:51 +0x2b1
github.com/openshift-pipelines/pipelines-as-code/pkg/matcher.RemoteTasks.GetTaskFromAnnotations({0xc000387ae0}, {0x1a9d690, 0xc000040128}, {0x1ac4f00, 0xc0001100a8}, 0x1)
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/matcher/annotation_tasks_install.go:79 +0x2e8
github.com/openshift-pipelines/pipelines-as-code/pkg/resolve.Resolve({0x1a9d690, 0xc000040128}, 0xc0004084e0, {0x1ac4f00, 0xc0001100a8}, {0xc000464000, 0x625}, 0xc00051dd00)
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/resolve/resolve.go:129 +0x215
github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/resolve.resolveFilenames(0x1633400, {0xc00042c450, 0x1878192, 0x9}, 0x0)
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/resolve/resolve.go:150 +0x11e
github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/resolve.Command.func1(0xc00042bb80, {0x1871c6f, 0x2, 0x2})
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/cmd/tknpac/resolve/resolve.go:103 +0x3d9
github.com/spf13/cobra.(*Command).execute(0xc00042bb80, {0xc0003e7520, 0x2, 0x2})
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc00042ac80)
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code/cmd/tkn-pac/main.go:14 +0xcb
```

--- a/pkg/matcher/annotation_tasks_install.go
+++ b/pkg/matcher/annotation_tasks_install.go
@@ -2,8 +2,10 @@ package matcher
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"regexp"
 	"strings"
 
@@ -12,6 +14,7 @@ import (
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider"
 	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
 	k8scheme "k8s.io/client-go/kubernetes/scheme"
 )
 
@@ -48,10 +51,23 @@ func (rt RemoteTasks) getTask(ctx context.Context, providerintf provider.Interfa
 		defer res.Body.Close()
 		return rt.convertTotask(string(data))
 	case strings.Contains(task, "/"):
-		data, err := providerintf.GetFileInsideRepo(ctx, rt.Run.Info.Event, task, "")
-		if err != nil {
-			return ret, err
+		var data string
+		var err error
+		if rt.Run.Info.Event.SHA != "" {
+			data, err = providerintf.GetFileInsideRepo(ctx, rt.Run.Info.Event, task, "")
+			if err != nil {
+				return ret, err
+			}
+		} else {
+			data, err = getTaskFromLocalFS(task, rt.Run.Clients.Log)
+			if err != nil {
+				return nil, err
+			}
+			if data == "" {
+				return nil, nil
+			}
 		}
+
 		return rt.convertTotask(data)
 	default:
 		data, err := hub.GetTask(ctx, rt.Run, task)
@@ -84,4 +100,24 @@ func (rt RemoteTasks) GetTaskFromAnnotations(ctx context.Context, providerintf p
 		}
 	}
 	return ret, nil
+}
+
+// getTaskFromLocalFS get task locally if file exist
+// TODO: may want to try chroot to the git root dir first as well if we are able so.
+func getTaskFromLocalFS(taskName string, logger *zap.SugaredLogger) (string, error) {
+	var data string
+	// We are most probably running with tkn pac resolve -f here, so
+	// let's try by any chance to check locally if the task is here on
+	// the filesystem
+	if _, err := os.Stat(taskName); errors.Is(err, os.ErrNotExist) {
+		logger.Warnf("could not find remote task %s inside repo", taskName)
+		return "", nil
+	}
+
+	b, err := ioutil.ReadFile(taskName)
+	data = string(b)
+	if err != nil {
+		return "", err
+	}
+	return data, nil
 }

--- a/pkg/matcher/annotation_tasks_install_test.go
+++ b/pkg/matcher/annotation_tasks_install_test.go
@@ -12,7 +12,11 @@ import (
 	httptesthelper "github.com/openshift-pipelines/pipelines-as-code/pkg/test/http"
 	"github.com/openshift-pipelines/pipelines-as-code/pkg/test/provider"
 	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	zapobserver "go.uber.org/zap/zaptest/observer"
 	"gotest.tools/v3/assert"
+	"gotest.tools/v3/env"
+	"gotest.tools/v3/fs"
 	k8scheme "k8s.io/client-go/kubernetes/scheme"
 	rtesting "knative.dev/pkg/reconciler/testing"
 )
@@ -45,6 +49,7 @@ spec:
 		runevent        info.Event
 		annotations     map[string]string
 		wantErr         string
+		wantLog         string
 		gotTaskName     string
 		remoteURLS      map[string]map[string]string
 		filesInsideRepo map[string]string
@@ -97,6 +102,9 @@ spec:
 			filesInsideRepo: map[string]string{
 				"be/healthy": simpletask,
 			},
+			runevent: info.Event{
+				SHA: "007",
+			},
 		},
 		{
 			name: "test-annotations-remote-http-skipping-notmatching",
@@ -125,6 +133,16 @@ spec:
 				pipelinesascode.GroupName + "/task": "[pas/la]",
 			},
 			wantErr: "could not find",
+			runevent: info.Event{
+				SHA: "007",
+			},
+		},
+		{
+			name: "test-annotations-remote-no-event-not-found-no-error",
+			annotations: map[string]string{
+				pipelinesascode.GroupName + "/task": "[not/here]",
+			},
+			wantLog: "could not find remote task not/here inside repo",
 		},
 		{
 			name:        "test-get-from-hub-latest",
@@ -164,9 +182,12 @@ spec:
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			httpTestClient := httptesthelper.MakeHTTPTestClient(t, tt.remoteURLS)
+			observer, fakelog := zapobserver.New(zap.InfoLevel)
+			logger := zap.New(observer).Sugar()
 			cs := &params.Run{
 				Clients: clients.Clients{
 					HTTP: *httpTestClient,
+					Log:  logger,
 				},
 				Info: info.Info{
 					Event: &tt.runevent,
@@ -187,7 +208,9 @@ spec:
 				assert.ErrorContains(t, err, tt.wantErr, "We should have get an error with %v but we didn't", tt.wantErr)
 				return
 			}
-
+			if tt.wantLog != "" {
+				assert.Assert(t, len(fakelog.FilterMessage(tt.wantLog).TakeAll()) > 0, "could not find log message: got ", fakelog)
+			}
 			assert.NilError(t, err, "GetTaskFromAnnotations() error = %v, wantErr %v", err, tt.wantErr)
 			assert.Assert(t, len(got) > 0, "GetTaskFromAnnotations() error no tasks has been processed")
 
@@ -196,4 +219,14 @@ spec:
 			}
 		})
 	}
+}
+
+func TestGetTaskFromLocalFS(t *testing.T) {
+	content := "hellomoto"
+	defer env.ChangeWorkingDir(t, fs.NewDir(t, "TestGetTaskFromLocalFS", fs.WithFile("task1", content)).Path())()
+	observer, _ := zapobserver.New(zap.InfoLevel)
+	logger := zap.New(observer).Sugar()
+	taskContent, err := getTaskFromLocalFS("task1", logger)
+	assert.NilError(t, err)
+	assert.Equal(t, content, taskContent)
 }

